tests/tests_api.c

```diff
@@ -1647,22 +1673,15 @@ static void run_utests(void)
 }
 
 
-uint32_t __stack_chk_guard = 0;
 
-extern void __attribute__((noreturn)) __stack_chk_fail(void);
-void __attribute__((noreturn)) __stack_chk_fail(void)
-{
-    printf("\n\nError: stack smashing detected!\n\n");
-    abort();
-}
 
 
 int main(void)
 {
     // Test the C code API
     TEST_LIVE_DEVICE = 0;
     random_init();
-    __stack_chk_guard = random_uint32(0);
+
```

Comment:
@jonasschnelli why do you remove the stack guard?

Reply:
Indeed, shouldn't be removed.

Reply:
The stack protector guard should be initialized by `libssp`

Reply:
Correct me if I'm wrong, but the initialization by libssp would use a fixed value (since it does not have access to an entropy source). Then setting it to a random value manually on startup using the HRNG would provide better security.

Reply:
Doesn't this code run on a computer? On Linux, `libssp` uses `/dev/urandom`.

Reply:
Oh yea. Wasn't paying attention to the file title... That's correct then.